File: solutions/171.Excel_Sheet_Column_Number/AC_simulation_1.py
# Author:      illuz <iilluzen[at]gmail.com>
# File:        AC_simulation_1.py
# Create Date: 2015-02-05 10:08:01
# Descripton:  Use lambda

class Solution:
    # @param s, a string
    # @return an integer
    def titleToNumber(self, s):
        return reduce(lambda x, y: 26 * x + ord(y) - ord('A') + 1, s, 0)

# debug
s = Solution()
print s.titleToNumber('AA')

File: solutions/168.Excel_Sheet_Column_Title/AC_simulation_1.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_simulation_1.cpp
*  Create Date: 2015-02-05 09:26:31
*  Descripton:  Simulation
*/

#include <bits/stdc++.h>

using namespace std;
const int N = 0;

class Solution {
public:
    string convertToTitle(int n) {
        string ret = "";
        --n;
        while (n >= 0) {
            ret += (char)('A' + n % 26);
            n /= 26;
            --n;
        }
        reverse(ret.begin(), ret.end());
        return ret;
    }
};

int main() {
    int n;
    Solution s;
    while (cin >> n)
        cout << s.convertToTitle(n) << endl;
    return 0;
}

